### DISCUSSION TITLE
How do I define a number as a pure intger in readprec 1 mode

### DISCUSSION THREAD

**Discussion by CharlesBClarke** [Qalculate/libqalculate#631] [Q&A] [answered]
I use readprec 1 while doing phyics
I was trying to find the Centripetal force on a proton using 
(proton_mass*((1.60E6m/s)^2+-0))/(5cm)
but I got the error
```
  warning: Calculation of uncertainty propagation failed (using interval arithmetic instead).
  ≈ (ProtonMass × ((((1.6000±0.0050 × (10^6)) meters/second)^2.00±0.50) − 0.00±0.50)) / (5.00±0.50 centimeters) ≈ 3.38E−23±0.34E−23((1.6E12±1.7E12 kg·m^10.00E−1±5.00E−1) / s^2.00±0.50 + (0.000 kg) / m)
```
The issue is clear ^2 should not become ^2.00±0.50, its not a measurement and should be infinity precise.

**Comment by CharlesBClarke** [comment]
I changed to readprec 2 and now I get this error
```
> ( 2*elementary_charge 35.6 km/s 1.80T)/(6.64e-27 kg)

  ≈ ((2 ×  ElementaryCharge × 35.600±0.050(kilometer/second) × 1.8000±0.0050 teslas) / ((6.6400±0.0050 × (10^−27)) kilograms)) ≈ 3.1E12 m/s²
```
The two causes the whole thing to be infinitely precise.  Here is the result without the two...
```
> ( elementary_charge 35.6 km/s 1.80T)/(6.64e-27 kg)

  ≈ (ElementaryCharge × 35.600±0.050(kilometer/second) × 1.8000±0.0050 teslas) / ((6.6400±0.0050 × (10^−27)) kilograms) ≈ 1.5462E12±0.0050E12 m/s²

```

**Comment by hanna-kn** [comment, marked as answer]
> The two causes the whole thing to be infinitely precise.

By default significant digits are shown without `±`, unless the number of significant digits are less than 2, or the input contains `±`. This can be changed using the "interval display" setting. If the output were more precise `≈` would have been replaced by '=' (for exact value), or zeroes appended to the number.

```
> help interval display

interval display (ivdisp)
(0* = adaptive, 1 = significant, 2 = interval, 3 = plusminus, 4 = midpoint, 5 = lower,
6 = upper, 7 = concise, 8 = relative)
```

**Reply by CharlesBClarke** [reply]
Thank you again for the help. One last question this helps me in readprec 2, is there a way I can use readperc 1 and define specific numbers as scalars? I want (5mm)^2 to become (5.00 ±0.50mm)^2 not (5.00 ±0.50mm)^(2.00±0.5)?

**Comment by hanna-kn** [comment]
> Thank you again for the help. One last question this helps me in readprec 2, is there a way I can use readperc 1 and define specific numbers as scalars? I want (5mm)^2 to become (5.00 ±0.50mm)^2 not (5.00 ±0.50mm)^(2.00±0.5)?

No, but you can use `readprec 1`, or `readprec 0`, and enter `(5+/-0.5 mm)^2`. Perhaps Qalculate should be smart enough to recognize that the exponents for values with units should reasonable never contain uncertainties.
